connectors/sources/mssql.py

```diff
@@ -0,0 +1,126 @@
+    "TABLE_DATA": 'SELECT * FROM {schema}."{table}"',
+    "TABLE_LAST_UPDATE_TIME": "SELECT last_user_update FROM sys.dm_db_index_usage_stats WHERE object_id=object_id('{schema}.{table}')",
+    "TABLE_DATA_COUNT": 'SELECT COUNT(*) FROM {schema}."{table}"',
+    "ALL_SCHEMAS": "SELECT s.name from sys.schemas s inner join sys.sysusers u on u.uid = s.principal_id",
+}
+
+
+class MSSQLDataSource(GenericBaseDataSource):
+    """Microsoft SQL Server"""
+
+    def __init__(self, configuration):
+        """Setup connection to the Microsoft SQL database-server configured by user
+
+        Args:
+            configuration (DataSourceConfiguration): Object of DataSourceConfiguration class.
+        """
+        super().__init__(configuration=configuration)
+        self.mssql_driver = self.configuration["mssql_driver"]
+        self.secured_connection = self.configuration["secured_connection"]
+        self.connection_string = ""
+        self.queries = QUERIES
+        self.dialect = "Microsoft SQL"
+
+    @classmethod
+    def get_default_configuration(cls):
+        """Get the default configuration for database-server configured by user
+
+        Returns:
+            dictionary: Default configuration
+        """
+        mssql_configuration = super().get_default_configuration().copy()
+        mssql_configuration.update(
+            {
+                "mssql_driver": {
+                    "value": "ODBC Driver 18 for SQL Server",
+                    "label": "Microsoft SQL Driver",
+                    "type": "str",
+                },
+                "secured_connection": {
+                    "value": SECURED_CONNECTION,
+                    "label": "Connection will be secured or not",
+                    "type": "bool",
+                },
+            }
+        )
+        return mssql_configuration
+
+    def _create_engine(self):
+        """Create sync engine for mssql"""
+        if self.secured_connection:
+            self.connection_string = URL.create(
+                "mssql+pyodbc",
+                username=self.user,
+                password=self.password,
+                host=self.host,
+                port=self.port,
+                database=self.database,
+                query={
+                    "driver": self.mssql_driver,
+                    "TrustServerCertificate": "no",
+                    "Encrypt": "Yes",
+                },
+            )
+        else:
+            self.connection_string = URL.create(
+                "mssql+pyodbc",
+                username=self.user,
+                password=self.password,
+                host=self.host,
+                port=self.port,
+                database=self.database,
+                query={"driver": self.mssql_driver, "TrustServerCertificate": "yes"},
+            )
+        self.engine = create_engine(self.connection_string)
```

Comment:
Why not create an engine in the initializer?

Reply:
As we need to pass the connection string to create the engine, we need to call a separate method inside `__init__()` and just to follow best practice not to call any complex method inside `__init__()` we created this method. However, we can move it to `__init__()` if you want. (for this and other dialects as well)

Reply:
That is true - in case you don't want it in `__init__()`, you can make it a lazy property of the class so that it's initialised once it's used.

Reply:
I have added the change in #565.